std/algorithm/iteration.d

```diff
@@ -916,12 +923,20 @@ template each(alias pred = "a")
         debug(each) pragma(msg, "Using foreach for ", Iterable.stringof);
         static if (isForeachUnaryIterable!Iterable)
         {
+            debug(each) pragma(msg, "unary");
             foreach (ref e; r)
+            {
                 cast(void)unaryFun!pred(e);
+            }
         }
-        else // if (isForeachBinaryIterable!Iterable)
+        else static if (isForeachBinaryIterable!Iterable)
         {
-            foreach (ref i, ref e; r)
+            foreach (ref a, ref b; r)
+                cast(void)binaryFun!BinaryArgs(a, b);
+        }
+        else // static if (isForeachUnaryWithIndexIterable!Iterable)
```

Comment:
Oh I opted for this because it was used like this before. I prefer `static assert` as well :)